toolchain/check/eval.cpp

```diff
@@ -769,6 +766,15 @@ static auto DiagnoseDivisionByZero(Context& context, SemIRLoc loc) -> void {
   context.emitter().Emit(loc, CompileTimeDivisionByZero);
 }
 
+// Get an integer at a suitable bit-width: either its actual width if it has a
+// fixed width, or the canonical width from the value store if not.
+static auto GetIntAtSuitableWidth(Context& context, IntId bit_width_id,
```

Comment:
Hm. I think the need to deal with the "unknown width" case in exactly this way is probably specific to the evaluation logic, and other users of `IntStore` are unlikely to want it -- especially in something with such an innocuous name :) It looks like all the current calls to `GetAtWidth` are in this file, but I think that's just because lower/constant.cpp hasn't been updated to use it yet -- [this](https://github.com/carbon-language/carbon-lang/blob/5169a1862ec34fd5e531b5f96e5b6d5988ed3fae/toolchain/lower/constant.cpp#L195) is precisely `GetAtWidth` written longhand, and doesn't want this special casing for an invalid width.